common/Unit.cpp

```diff
@@ -176,10 +176,13 @@ void UnitWSD::lookupTile(int part, int width, int height, int tilePosX, int tile
                          int tileWidth, int tileHeight,
                          std::shared_ptr<std::vector<char>> &tile)
 {
-    if (tile)
-        onTileCacheHit(part, width, height, tilePosX, tilePosY, tileWidth, tileHeight);
-    else
-        onTileCacheMiss(part, width, height, tilePosX, tilePosY, tileWidth, tileHeight);
+    if (isUnitTesting())
```

Comment:
Would have been better to find out what was segfaulting. I suspect something not initialized on Android or something silly like that. What does the debugger show at the point of segfault ?